external/tools/modify_imgui_for_gearsvk.ps1

```diff
@@ -0,0 +1,292 @@
+# 	vkCmdBindDescriptorSets(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL);'
+#
+# This changes:
+# - The comment "// Bind pipeline and descriptor sets:" in ImGui_ImplVulkan_SetupRenderState to:
+#	// Bind pipeline
+#
+# - The line "io.Fonts->SetTexID((ImTextureID)(intptr_t)g_FontImage);" in ImGui_ImplVulkan_CreateFontsTexture for setting the font texture id to:
+#	ImTextureID texID = (ImTextureID)g_DescriptorSet;
+#	io.Fonts->SetTexID(texID);
+#
+# This removes:
+# - User texture binding as missing feature comment at the top:
+#	//  [ ] Renderer: User texture binding
+#
+# - The unecessary font texture descriptor binding:
+#	VkDescriptorSet desc_set[1] = { g_DescriptorSet };
+#	vkCmdBindDescriptorSets(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL);
+##
+
+"- Modifying " + $imgui_impl_vulkan_cpp + ":"
+
+# Read imgui vulkan impl source
+$fileContent = Get-Content $imgui_impl_vulkan_cpp
+
+# Helpers for function detection
+$inSetupRenderState = 0
+$inRenderDrawData = 0
+$inCreateFontsTexture = 0
+
+# Create an empty array and use it as the modified file
+$fileContentModified = @()
+
+# Header in file to communicate modification and prevent
+# multiple executions of this script
+$gearsVkHeader = "// Modified version for Gears-Vk"
+$fileContentModified += $gearsVkHeader
+$fileContentModified += '// Additions:'
+$fileContentModified += '// - User texture binding and descriptor set caching'
+$fileContentModified += ''
+
+$fileAlreadyModified = 0
+Foreach ($line in $fileContent)
+{
+	# Skip file if it already contains our header
+	if ($line -match $gearsVkHeader) {
+		Write-Host " -> already modified" -ForegroundColor Green
+		$fileAlreadyModified = 1
+		break
+	}
+
+	# Add user texture binding to implemented features list
+	if ($line -match "// Missing features:") {
+		$fileContentModified += "//  [x] Renderer: User texture binding. ImTextureID is used to store a handle to a Descriptorset."
+	}
+	# Remove user texture binding comment from missing feature
+	if ($line -match "//  \[ \] Renderer: User texture binding") {
+		continue
+	}
+
+	# Detect if in ImGui_ImplVulkan_SetupRenderState
+	if ($line -match "^static void ImGui_ImplVulkan_SetupRenderState\(") {
+		$inSetupRenderState = 1
+		Write-Host "  - Modifying ImGui_ImplVulkan_SetupRenderState"
+	}
+
+	# Detect if in ImGui_ImplVulkan_RenderDrawData
+	if ($line -match "^void ImGui_ImplVulkan_RenderDrawData\(") {
+		$inRenderDrawData = 1
+		Write-Host "  - Modifying ImGui_ImplVulkan_RenderDrawData"
+	}
+
+	# Detect if in ImGui_ImplVulkan_CreateFontsTexture
+	if ($line -match "^bool ImGui_ImplVulkan_CreateFontsTexture\(") {
+		$inCreateFontsTexture = 1
+		Write-Host "  - Modifying ImGui_ImplVulkan_CreateFontsTexture"
+	}
+
+	# Detect if the function we have modified is left by relying on consistent code
+	# formatting, i.e. the closing '}' is at the beginning of the line
```

Comment:
Hmm... that's a bit dangerous. Optimally, one would count opening and closing curly braces per line and make a decision based on that. At least we should add some white-space safety checks.